examples/json/lib/functions.dart

```diff
@@ -17,6 +17,25 @@ import 'package:json_annotation/json_annotation.dart';
 
 part 'functions.g.dart';
 
+@JsonSerializable()
+class GreetingRequest {
+  final String? name;
+
+  GreetingRequest({this.name});
+
+  factory GreetingRequest.fromJson(Map<String, dynamic> json) =>
+      _$GreetingRequestFromJson(json);
+
+  Map<String, dynamic> toJson() => _$GreetingRequestToJson(this);
+
+  @override
+  bool operator ==(Object other) =>
+      other is GreetingRequest && other.name == name;
+
+  @override
+  int get hashCode => name.hashCode;
```

Comment:
Are we comparing these anywhere? Just curious if we can keep the complexity down...